src/components/Clock/Clock.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export class Clock extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: 'React clock',
+    time: new Date(),
+  }
+
+  componentDidMount() {
+    if (this.state.isClockVisible) {
+      this.timer();
+    }
+  }
+
+  timer() {
+    setInterval(() => {
```

Comment:
you need to be sure that on unmounting your component all intervals are removed, add componentWillUnmount method